agent/job_runner.go

```diff
@@ -435,6 +467,26 @@ func (r *JobRunner) createEnvironment(ctx context.Context) ([]string, error) {
 		env["BUILDKITE_IGNORED_ENV"] = strings.Join(ignoredEnv, ",")
 	}
 
+	// Check if allowed environment variables are set
+	if len(r.conf.AgentConfiguration.AllowedEnvironmentVariables) > 0 {
+		// Fetch environment variables that are allowed by default
+		var allowedEnv = make([]string, len(DefaultAllowedEnv))
+		copy(allowedEnv, DefaultAllowedEnv)
+
+		var deniedEnv []string
+
+		// And add allowed environment variables from agent configuration
+		allowedEnv = append(allowedEnv, r.conf.AgentConfiguration.AllowedEnvironmentVariables...)
+
+		// Filter to get denied envs
+		env, deniedEnv = filterEnv(env, allowedEnv)
```

Comment:
Are you purposely overwriting that env?
```suggestion
		allowedEnvToValues, deniedEnv := filterEnv(env, allowedEnv)
```

Reply:
Yup, it needs to be overwritten as the function doesn't edit `env` but creates a new `map` with filtered environment variables.